providers/implementations/kdfs/hkdf.c

```diff
@@ -262,8 +260,12 @@ static int kdf_hkdf_get_ctx_params(void *vctx, OSSL_PARAM params[])
     KDF_HKDF *ctx = (KDF_HKDF *)vctx;
     OSSL_PARAM *p;
 
-    if ((p = OSSL_PARAM_locate(params, OSSL_KDF_PARAM_SIZE)) != NULL)
-        return OSSL_PARAM_set_size_t(p, kdf_hkdf_size(ctx));
+    if ((p = OSSL_PARAM_locate(params, OSSL_KDF_PARAM_SIZE)) != NULL) {
+        size_t sz = kdf_hkdf_size(ctx);
+        int ret = OSSL_PARAM_set_size_t(p, sz);
```

Comment:
I think this would be better coded to avoid the set on error.

i.e.
```c
size_t sz = kdf_hkdf_size(ctx);

if (sz == 0)
    return 0;
return OSSL_PARAM_set_size_t(p, kdf_hkdf_size(sz));
```

It's a line longer but easier to understand (hence no comment).